sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -242,10 +242,20 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     ctx match {
       case table: InsertIntoTableContext =>
         val (tableIdent, partitionKeys, exists) = visitInsertIntoTable(table)
-        InsertIntoTable(UnresolvedRelation(tableIdent), partitionKeys, query, false, exists)
+        InsertTableStatement(
+          UnresolvedRelation(tableIdent),
+          partitionKeys,
+          query,
+          overwrite = false,
+          exists)
```

Comment:
Can we rename this to `ifNotExists`? I think the other name isn't clear enough and this is a good opportunity to fix it.

Reply:
#18050 renamed parameter `ifNotExists` to `ifPartitionNotExists` for `InsertIntoTable`. Should we use `ifPartitionNotExists`?